### PR TITLE
Remove desugaring

### PR DESCRIPTION
**Pull request by chriseth**
This is in preparation for unifying the documentation.

### CODE REVIEW

File: docs/assembly.rst [hunk 1]
```diff
@@ -418,6 +418,9 @@ changes during the call, and thus references to local variables will be wrong.
 Labels
 ------
 
+.. note::
+    Labels are deprecated. Please use functiosn, loops, if or switch statements instead.
```
Comment (axic) [inline docs/assembly.rst]: `functiosn` typo

File: docs/assembly.rst [hunk 2]
```diff
@@ -693,9 +700,9 @@ the form ``mul(add(x, y), 7)`` are preferred over pure opcode statements like
 ``7 y x add mul`` because in the first form, it is much easier to see which
 operand is used for which opcode.
 
-The second goal is achieved by introducing a desugaring phase that only removes
-the higher level constructs in a very regular way and still allows inspecting
-the generated low-level assembly code. The only non-local operation performed
+The second goal is achieved by compiling the higher
+higher level constructs to bytecode in a very regular way.
```
Comment (axic) [inline docs/assembly.rst]: double `higher`

Comment (chriseth) [reply docs/assembly.rst]: changed.

File: docs/assembly.rst (hunk 1 above)
Comment (chriseth) [reply docs/assembly.rst]: changed.

**Review by axic — APPROVED** [review]

**Pull request merged**